### PR TITLE
Added deprecation notice for curl builder.

### PR DESCRIPTION
**Pull request by bendory**
https://github.com/GoogleCloudPlatform/cloud-builders/issues/672

### CODE REVIEW

File: curl/README.md [hunk 1]
```diff
@@ -16,7 +17,7 @@ file must be publicly readable, since no credentials are passed in the request.
 
 ```
 steps:
-- name: gcr.io/cloud-builders/curl
+- name: 'curlimages/curl'
   args: ['http://www.example.com/remotefile.zip', '--output', 'localfile.zip']
```
Comment (leeonlee) [inline curl/README.md:21]: This doesn't work because the new curl image runs as a non root user and can't write locally

Comment (bendory) [reply curl/README.md:21]: ewww...

I'll update the example to one that works...

File: curl/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,3 @@
-FROM launcher.gcr.io/google/ubuntu16_04
-
-RUN apt-get update && \
```
Comment (bendory) [inline curl/Dockerfile:3]: The base image already has both `curl` and `ca-certificates` installed.

File: curl/README.md [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
-Arguments passed to this builder will be passed to `curl` directly.
+This image is deprecated.
+
+For best support of `curl` please use one of the official `curl` images
+maintained by the `curlimages` community on Dockerhub.
+
+For details, visit https://hub.docker.com/r/curlimages/curl.
+
+Note that `curlimages/curl` executes as special user `curl_user` and thus may
+not echo be suitable for all purposes.
```
Comment (sanastos) [inline curl/README.md:13]: `not echo be suitable`

Comment (bendory) [reply curl/README.md]: good catch, thanks!

File: curl/deprecation.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+echo           \*\*\*\*\* DEPRECATION NOTICE \*\*\*\*\*
+echo
+echo This image is deprecated and will no longer be updated.
+echo This recent version of the image will continue to exist.
+echo
+echo For best support of \`curl\` please use one of the official
+echo \`curl\` images maintained by the curlimages community
+echo on Dockerhub.
+echo For details, visit https://hub.docker.com/r/curlimages/curl. Note that
+echo this image executes as special user \`curl_user\` and thus may not echo be
```
Comment (leeonlee) [inline curl/deprecation.sh:12]: There's an 'echo' out of place in this line

Comment (bendory) [reply curl/deprecation.sh]: thanks, fixed

**Pull request merged**